gazebo_grasp_plugin/src/GazeboGraspFix.cpp

```diff
@@ -740,60 +788,97 @@ void GazeboGraspFix::OnContact(const ConstContactsPtr &_msg)
+#else
+            // now, get average pose relative to the colliding link
+            gz_math_Pose3d linkWorldPose=linkCollision->GetLink()->GetWorldPose();
+
+            // To find out the collision point relative to the Link's local coordinate system, first get the Poses as 4x4 matrices
+            gz_math_Matrix4d worldToLink=linkWorldPose.rot.GetAsMatrix4();
+            worldToLink.SetTranslate(linkWorldPose.pos);
+
+            gz_math_Matrix4d worldToContact=gz_math_Matrix4d::IDENTITY;
             //we can assume that the contact has identity rotation because we don't care about its orientation.
             //We could always set another rotation here too.
             worldToContact.SetTranslate(avgPos);
 
             // now, worldToLink * contactInLocal = worldToContact
             // hence, contactInLocal = worldToLink.Inv * worldToContact
-            gazebo::math::Matrix4 worldToLinkInv = worldToLink.Inverse();
-            gazebo::math::Matrix4 contactInLocal = worldToLinkInv * worldToContact;
-            gazebo::math::Vector3 contactPosInLocal = contactInLocal.GetTranslation();
+            gz_math_Matrix4d worldToLinkInv = worldToLink.Inverse();
+            gz_math_Matrix4d contactInLocal = worldToLinkInv * worldToContact;
+            gz_math_Vector3d contactPosInLocal = contactInLocal.GetTranslation();
+
+#endif
             
             //std::cout<<"---------"<<std::endl;    
             //std::cout<<"CNT in loc: "<<contactPosInLocal.x<<","<<contactPosInLocal.y<<","<<contactPosInLocal.z<<std::endl;
 
-            /*gazebo::math::Vector3 sDiff=avgPos-linkWorldPose.pos;
+            /*gz_math_Vector3d sDiff=avgPos-linkWorldPose.pos;
             std::cout<<"SIMPLE trans: "<<sDiff.x<<","<<sDiff.y<<","<<sDiff.z<<std::endl;
             std::cout<<"coll world pose: "<<linkWorldPose.pos.x<<", "<<linkWorldPose.pos.y<<", "<<linkWorldPose.pos.z<<std::endl; 
             std::cout<<"contact avg pose: "<<avgPos.x<<", "<<avgPos.y<<", "<<avgPos.z<<std::endl; 
 
-            gazebo::math::Vector3 lX=linkWorldPose.rot.GetXAxis();    
-            gazebo::math::Vector3 lY=linkWorldPose.rot.GetYAxis();    
-            gazebo::math::Vector3 lZ=linkWorldPose.rot.GetZAxis();    
+            gz_math_Vector3d lX=linkWorldPose.rot.GetXAxis();
+            gz_math_Vector3d lY=linkWorldPose.rot.GetYAxis();
+            gz_math_Vector3d lZ=linkWorldPose.rot.GetZAxis();
     
             std::cout<<"World ori: "<<linkWorldPose.rot.x<<","<<linkWorldPose.rot.y<<","<<linkWorldPose.rot.z<<","<<linkWorldPose.rot.w<<std::endl;
             std::cout<<"x axis: "<<lX.x<<","<<lX.y<<","<<lX.z<<std::endl;
             std::cout<<"y axis: "<<lY.x<<","<<lY.y<<","<<lY.z<<std::endl;
             std::cout<<"z axis: "<<lZ.x<<","<<lZ.y<<","<<lZ.z<<std::endl;*/
 
+#if GAZEBO_MAJOR_VERSION >= 8
             // now, get the pose of the object and compute it's relative position to the collision surface.
```

Comment:
Same with getters and setters as commented above. Basically this code here should be free of #if.